proposals/0000-exception-backtraces.rst

```diff
@@ -224,6 +345,116 @@ the fact that stack traces are primarily a debugging tool and somewhat of a
 cross-cutting concern. While a stateless approach would be preferred, we
 believe that this compromise is a significant improvement over the status quo.
 
+The Haskell community will have to adapt it's code to the new exception structure.
+As described in `Adding ``SomeExceptionWithBacktrace```_ the expected impact isn't
+very high. This is further discussed in `Migration`_ .
+
+Migration
+---------
+
+There was an intense discussion in the comments of the pull request of this
+proposal about how to reach two goals
+(<https://github.com/ghc-proposals/ghc-proposals/pull/330>):
+- Keep the migration costs as low as possible. I.e. most usages should work
+  without any change.
+- Get a type checking error when the changed exception structure breaks existing
+  code. (I.e. no silent changes in behaviour.)
+
+The solution presented in this proposal has been agreed upon by all involved
+parties.
+
+``catch``and ``handle`` work with both, ``SomeExceptionWithBacktrace`` and
+``SomeException``. This is the main reason for keeping ``SomeException``
+as a layer in the exception hierarchy.
+
+The usages of ``throw`` and ``throwIO`` don't have to be changed, too.
+In fact, it showed that most submodules of GHC don't need any changes.
+Only a few changes were needed to be made to GHC itself (e.g. in ``compiler/``).
+
+Direct calls to ``toException`` and ``fromException``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As the types of ``toException`` and ``fromException`` changed, calls to them might
+need to be adjusted to the new types. This is guided by type checking errors.
+
+Usually, when these calls aren't bound to ``SomeException`` specific types, no
+change should be needed.
+
+In expressions where ``fromException``'s changed parameter type leads to type 
+errors, one common solution is to convert the exception value first with
+``toException``.
+
+As an example let's consider we don't want to change the type of 
```

Comment:
```suggestion
As an example let's consider the case of GHC's internal `GHC.TopHandler.real_handler` function; to ease compatibility, we may want to avoid changing the `SomeException` argument to `SomeExceptionWithBacktrace`. We can achieve this with a strategically-placed `fromException . toException`:
```